scipy/signal/windows.py

```diff
@@ -1065,6 +1066,81 @@ def kaiser(M, beta, sym=True):
     return w
 
 
+def kaiser_derived(M, beta=4.):
+    """ Return a Kaiser-Bessel derived window.
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an empty
+        array is returned.
+    beta : float
+        Kaiser-Bessel window shape parameter.
+
+    Returns
+    -------
+    w : ndarray
+        The window, with the maximum value normalized to 1.
+
+    Notes
+    -----
+    This window is only defined for an even number of taps.
+
+    .. versionadded:: 0.16.0
+
+    References
+    ----------
+    .. [1] Wikipedia, "Kaiser window",
+           https://en.wikipedia.org/wiki/Kaiser_window
+
+    Examples
+    --------
+    Plot the window and its frequency response:
+
+    >>> from scipy import signal
+    >>> from scipy.fftpack import fft, fftshift
+    >>> import matplotlib.pyplot as plt
+
+    >>> window = signal.kaiser_derived(50)
+    >>> plt.plot(window)
+    >>> plt.title("Kaiser-Bessel derived window")
+    >>> plt.ylabel("Amplitude")
+    >>> plt.xlabel("Sample")
+
+    >>> plt.figure()
+    >>> A = fft(window, 2048) / (len(window)/2.0)
+    >>> freq = np.linspace(-0.5, 0.5, len(A))
+    >>> response = 20 * np.log10(np.abs(fftshift(A / abs(A).max())))
+    >>> plt.plot(freq, response)
+    >>> plt.axis([-0.5, 0.5, -120, 0])
+    >>> plt.title("Frequency response of the kaiser bessel derei window")
+    >>> plt.ylabel("Normalized magnitude [dB]")
+    >>> plt.xlabel("Normalized frequency [cycles per sample]")
+    >>> plt.show()
+
+    """
+    if M < 1:
+        return np.array([])
+    if M == 1:
+        return np.ones(1, 'd')
```

Comment:
I hope they "fall back" to returning 1 because that is the correct (if somewhat degenerate) value to use for the window, and not because they are arbitrarily choosing to do so.

In this case, it is inconsistent to say that the length of the window must be even, but then allow the length to be 1.  How about just removing this check, and letting it raise a ValueError when M is 1?